src/lib.rs

```diff
@@ -907,109 +907,124 @@ unsafe fn close_internal(c_out: &mut *mut Rav1dContext, flush: c_int) {
     if flush != 0 {
         rav1d_flush(c);
     }
-    if !((*c).tc).is_null() {
-        let ttd: *mut TaskThreadData = &mut (*c).task_thread;
-        if (*ttd).inited != 0 {
-            let task_thread_lock = (*ttd).delayed_fg.lock().unwrap();
-            let mut n: c_uint = 0 as c_int as c_uint;
-            while n < (*c).n_tc && (*((*c).tc).offset(n as isize)).task_thread.td.inited != 0 {
-                (*((*c).tc).offset(n as isize)).task_thread.die = true;
-                n = n.wrapping_add(1);
+    c.drop_in_place();
+    rav1d_freep_aligned(c_out as *mut _ as *mut c_void);
+}
+
+impl Drop for Rav1dContext {
+    fn drop(&mut self) {
+        unsafe {
```

Comment:
Can you leave a safety comment saying this isn't probably safe yet (which we'll fix eventually), but we need it here since it's in a trait?